lib/luahid.c

```diff
@@ -153,6 +173,64 @@ static int luahid_probe(struct hid_device *hdev, const struct hid_device_id *id)
 	return hid_hw_start(hdev, HID_CONNECT_DEFAULT);
 }
 
+static int luahid_doreport_fixup(lua_State *L, luahid_t *hid,
+				 struct hid_device *hdev, __u8 *buf, unsigned int *size,
+				 __u8 **ret_ptr)
+{
+	if (luahid_checkdriver(L, hid, -1, "_report_des")) {
+		pr_err("report_fixup: couldn't find driver\n");
+		return -ENXIO;
+	}
```

Comment:
this can be turned into a inline that returns 0 or -ENXIO, as you are using this pattern multiple times..